draft-ietf-httpbis-sfbis.md

```diff
@@ -474,14 +474,17 @@ Note that in Dictionary ({{dictionary}}) and Parameter ({{param}}) values, Boole
 
 Date values can be conveyed in Structured Fields.
 
-Dates have a data model that is similar to Integers, representing a (possibly negative) delta in seconds from January 1, 1970 00:00:00 UTC, excluding leap seconds.
+Dates have a data model that is similar to Integers, representing a (possibly negative) delta in seconds from 1 January 1970 00:00:00 UTC, excluding leap seconds.
 
 For example:
 
 ~~~ http-message-new
 Example-Date: @1659578233
 ~~~
 
+Parsers MUST support Dates whose values include all days in years 1 to 9999 (i.e., -62,135,596,800 to 253,402,214,400 delta seconds from 1 January 1970 00:00:00 UTC).
```

Comment:
I dont like this.

We have specified a serialization that allows +/- 15 digits and any parser which does not grok that is non-compliant, so this guidance interval just adds confusion.

Reply:
Right, but in the same way, we specify that implementations must support strings of at least a certain length, etc. We have at least implementation languages that convert a date into a native object will _not_ support the full range of 15 digits, so it's good to set expectations there.

Reply:
Any more thoughts @bsdphk - does that help?

Reply:
+1 to 4 digit years. That’s what we use for dates in RFC 9110, I do not think we want to have conversion errors between the textual notation that we would otherwise use.

Reply:
Well. What we specify here is not just for HTTP dates.

I'm actually with @bsdphk here - keep it as simple as possible, and let the range be exactly the same as for integers.

Reply:
@reschke As stated by @mnot, there is concern of interoperability issues with that approach.

An example is the Date object of JavaScript, IIRC its range is IEEE754 in the unit of *milliseconds*.

Reply:
Yes. But that implies that you actually will convert to a date object, instead of just leaving it in integer form...

Reply:
This is a minimal requirement ("MUST support ...") so I think we'd want to be on the safe side to make sure that nothing unexpected happens even if they are converted to date objects, assuming that the sender stays in the stated range?

Reply:
Again, **we are not talking about the range of acceptable values.** We are talking about the minimum range an implementation is required to support. 

Just as some implementations might not support 32k-long Strings or Tokens, some implementations might clamp the range of Dates they support because they're trying to fit things into an idiomatic object or other representation. By stating these bounds, we assure users of the type that at least that range will be supported and interoperable. 

If we don't say this, the outcome will largely be the same -- some implementations will wrap Dates into objects that don't support dates 20,000 years into the future, or before year 0. The only function of the text is giving users of the type a heads-up about what they can rely upon.

Reply:
As I said previously, we are in bikeshed territory now, so I'll not add more colors.

...

Except, shouldn't line 487 also use ISO8601 ?  :-)

Reply:
Heh. Done.